dialogue-example/src/main/java/com/palantir/dialogue/example/SampleServiceClient.java

```diff
@@ -162,14 +175,7 @@ public static AsyncSampleService async(Channel channel, ConjureRuntime runtime)
                 Call call = channel.createCall(STRING_TO_STRING, request);
                 return Futures.transform(
                         Calls.toFuture(call),
-                        response -> {
-                            try {
-                                // TODO(rfink): The try/catch is a bit odd here.
-                                return sampleObjectToSampleObjectDeserializer.deserialize(response);
-                            } catch (IOException e) {
-                                throw new RuntimeException("Failed to deserialize response", e);
-                            }
-                        },
+                        response -> sampleObjectToSampleObjectDeserializer.deserialize(response),
```

Comment:
does this mean the async calls have no timeout?

Reply:
correct. you're handed a Future and are in charge of timing it out yourself via `.get(timeout)`

Reply:
we can revisit this design if you like, but it seems non-trivial to plumb the timeouts into the underlying http client, in particular for the java one.

Reply:
for instance, the java client doesn't distinguish between read and write timeouts while the okhttp client does